File: list_all_tables.js

import { createClient } from '@supabase/supabase-js';

const supabaseUrl = "https://lqdxsgnoeickomhsgeco.supabase.co";
const serviceRoleKey = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImxxZHhzZ25vZWlja29taHNnZWNvIiwicm9zZSI6InNlcnZpY2Vfcm9sZSIsImlhdCI6MTc2Nzg1MjAwMSwiZXhwIjoyMDgzNDI4MDAxfQ.rBmS_M_yhr6CDK4_B8LQ5DG3_z1xEc5UHU4qwtC0-Hc";

const supabase = createClient(supabaseUrl, serviceRoleKey);

async function listAllTables() {
    console.log("Listing all tables in 'public' schema...");
    const { data: tables, error } = await supabase.from('information_schema.tables').select('table_name').eq('table_schema', 'public');

    if (error) {
        console.error("Error querying information_schema:", error);
        // Try Postgres RPC if available
        const { data: tables2, error: e2 } = await supabase.rpc('get_table_names');
        if (e2) console.error("RPC failed too:", e2);
        else console.log("Tables (via RPC):", tables2);
    } else {
        console.log("Tables Found:");
        tables.forEach(t => console.log(`- ${t.table_name}`));
    }
}

listAllTables();

File: list_all_tables.py
import os
import sys
from dotenv import load_dotenv
from supabase import create_client, Client

load_dotenv(dotenv_path="backend/.env")

url = os.getenv("SUPABASE_URL")
key = os.getenv("SUPABASE_SERVICE_ROLE_KEY")

supabase: Client = create_client(url, key)

def list_tables():
    sql = """
    SELECT table_name 
    FROM information_schema.tables 
    WHERE table_schema = 'public'
    ORDER BY table_name;
    """
    try:
        # Note: exec_sql returns void, so we might not get the list back directly
        # Unless the exec_sql is defined differently.
        # Let's check the definition of exec_sql if possible.
        res = supabase.rpc("exec_sql", {"query": sql}).execute()
        print("Response from RPC:", res.data)
        
        # If exec_sql returns results (SET RECORD or JSON), we'll see them.
        # If it returns void, we won't.
    except Exception as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    list_tables()
